### PR TITLE
HDPI-82 - Add local redis cache

### PR DESCRIPTION
**Pull request by JakeCG**
### Jira link

[HDPI-82](https://tools.hmcts.net/jira/browse/HDPI-82)

### Change description

As an Infrastructure engineer

I want to add Redis to local setup for frontend

So that it aids devs in running application locally

### Testing done

Local redis cache added and auto pinged.

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/modules/session/index.ts [hunk 1]
```diff
@@ -9,8 +9,12 @@ const { Logger } = require('@hmcts/nodejs-logging');
 export class Session {
   logger = Logger.getLogger('session');
   enableFor(app: Express): void {
-    const redis = new Redis(config.get<string>('secrets.pcs.redis-connection-string'));
+    const redisConnectionString =
+      process.env.REDIS_CONNECTION_STRING || config.get<string>('secrets.pcs.redis-connection-string');
+    this.logger.info(`Connecting to Redis at: ${redisConnectionString}`);
```
Comment (adusumillipraveen) [inline src/main/modules/session/index.ts:14]: Do not log the connection string please

File: docker-compose.yml [hunk 2]
```diff
@@ -10,5 +7,34 @@ services:
...
+    depends_on:
+      pcs-frontend-cache:
+        condition: service_healthy
+    environment:
+      - REDIS_CONNECTION_STRING=redis://pcs-frontend-cache:6379
+    networks:
+      - frontend-network
+
+  pcs-frontend-cache:
+    image: bitnami/redis:7.4.1
```
Comment (adusumillipraveen) [inline docker-compose.yml:20]: We are on Azure Redis version 6, but i don't think it might cause any issues. @MoLala1 ?

Comment (MoLala1) [reply docker-compose.yml:20]: it technically shouldnt be an issue as in charts.yaml: https://github.com/hmcts/pcs-frontend/blob/1cf213dd7bf0529f7522fd62a606353cf6ea9615/charts/pcs-frontend/Chart.yaml#L11

am using image version 20.3.0 but please test to be sure Jake :-)

**Review by MoLala1 — APPROVED** [review]
approved

File: docker-compose.yml (hunk 2 above)
Comment (adusumillipraveen) [reply docker-compose.yml:20]: It's not an issue currently, but we are developing against a latest version in local and someone might make a change and it to realise it won't work in Azure as azure instances are o old version. But happy to try out and change if we see such issues in future.

File: src/main/modules/session/index.ts (hunk 1 above)
Comment (MoLala1) [reply src/main/modules/session/index.ts]: was only for testing as part of my initial implementation, which i clearly didnt remove :-)

File: src/main/modules/session/index.ts [hunk 3]
```diff
@@ -9,7 +9,9 @@ const { Logger } = require('@hmcts/nodejs-logging');
 export class Session {
   logger = Logger.getLogger('session');
   enableFor(app: Express): void {
-    const redis = new Redis(config.get<string>('secrets.pcs.redis-connection-string'));
+    const redisConnectionString =
+      process.env.REDIS_CONNECTION_STRING || config.get<string>('secrets.pcs.redis-connection-string');
```
Comment (adusumillipraveen) [inline src/main/modules/session/index.ts:13]: this should work as there is env var mapping already?

```suggestion
      config.get<string>('secrets.pcs.redis-connection-string');
```

**Review by adusumillipraveen — APPROVED** [review]

**Pull request merged**